src/main/scala/com/rawlabs/das/server/grpc/TableServiceGrpcImpl.scala

```diff
@@ -299,40 +323,64 @@ class TableServiceGrpcImpl(
   private def runStreamedResult(
       source: Source[Row, NotUsed],
       request: ExecuteTableRequest,
-      responseObserver: StreamObserver[Rows]): Unit = {
-
-    val chunked: Source[Seq[Row], NotUsed] = {
-      if (maxChunkSize > 1) {
-        source.grouped(maxChunkSize)
-      } else {
-        // If maxChunkSize is 1, we don't need to group
-        source.map(row => Seq(row))
+      responseObserver: StreamObserver[Rows],
+      maybeServerCallObs: Option[ServerCallStreamObserver[Rows]]) = {
+
+    // Define the maximum bytes per chunk
+    val clientMaxBytes = request.getMaxBatchSizeBytes * 3 / 4;
```

Comment:
Please add a "big fat comment" explaining the 3/4. I believe it's to compensate for grpc overhead/metadata.